### PR TITLE
Ticket gaining methods

### PR DESCRIPTION
**Pull request by Kjaz816**
Added ticket gaining methods

### CODE REVIEW

**Review by bondgirl100 — CHANGES_REQUESTED** [review]
Bit of an error coming up when we try to write a new high score due to : can only concatenate str (not "int") to str at line 46 in avatar.py
Also when I opened the game for the first time, the previous high scores haven't been cleared yet
Couple of changes after this should work well!

**Review by ananyaahluwalia01 — APPROVED** [review]
Looks good! Maybe make "MainGame/Avatar/ticketCount.txt" a variable instead of repeating the file path every time you use it. I think sonar cloud is suggesting the same - so could be good for less duplication. Other than that it looks good!

**Comment by bondgirl100** [comment]
Still getting an error thrown by the text file this time - might need further review or discussion before being merged

**Review by bondgirl100 — APPROVED** [review]

File: MainGame/Snake/snakePlayed.txt [hunk 1]
```diff
@@ -0,0 +1 @@
+1
```
Comment (cnat307) [inline MainGame/Snake/snakePlayed.txt:1]: I think when the user first opens the app, all the values for games played should be 0. It looks like snake is the only game where its default value isn't 0.

**Review by cnat307 — COMMENTED** [review]
All the given methods for gaining tickets within games are working great.
But when considering the "times played" stats within the high scores window, the space invaders game doesn't update after playing.

Also, It would be good to consider the **state of the app** when it closes/restarts.
The number of tickets the user has and the games they've played stay in memory. But if they unlock an avatar, it will consume their tickets, and if they restart the app, that avatar becomes locked again.
To make the idea of state more consistent, it would be good to either make it so things such as tickets and "times played" reset after closing the app, or have it so the avatars stay unlocked after restarting the app.

File: MainGame/Snake/snakePlayed.txt (hunk 1 above)
[the remark this answers is not in the document]
Comment (hjia095) [reply MainGame/Snake/snakePlayed.txt:1]: > > Currently all the avater related like, selected avatar, tickets, purchased avatar, they got reset when the game restart, maybe add something to store it locally would be good.
> 
> I thought about adding this and combining this issue with issue #77 . However, I decided to close issue 77 with the following reasoning: "Given that none of the other states are saved when you exit the game (e.g. avatar, tickets, levels), I propose that we treat each opening of the program as a new player. That way, states are saved during one session, but do not persist beyond that. The high scores are you competing against other players."

We discuessed this Acatar state problem in #87, according to this and to keep the state consistent, just make a full reset on all avatar and ticket related objects

**Review by cnat307 — APPROVED** [review]
LGTM

**Pull request merged**